src/CMakeLists.txt

```
@@ -17,59 +16,48 @@ if (USE_INTERNAL_URDF)
     urdf/urdf_parser/urdf_model_state.cpp
     urdf/urdf_parser/urdf_sensor.cpp
     urdf/urdf_parser/world.cpp)
-  target_include_directories(using_parser_urdf INTERFACE
-    ${CMAKE_CURRENT_SOURCE_DIR}/urdf)
-  if (WIN32)
-    target_compile_definitions(using_parser_urdf INTERFACE -D_USE_MATH_DEFINES)
-  endif()
-else()
-  target_link_libraries(using_parser_urdf INTERFACE
-    IgnURDFDOM::IgnURDFDOM)
+  set(sources ${sources} ${urdf_internal_sources})
 endif()
 
-if (BUILD_TESTING)
-  # Build this test file only if Ignition Tools is installed.
-  if (NOT IGN_PROGRAM)
-    list(REMOVE_ITEM gtest_sources ign_TEST.cc)
-  endif()
+ign_create_core_library(SOURCES ${sources}
+  CXX_STANDARD 17
+  LEGACY_PROJECT_PREFIX SDFormat
+  )
 
-  # Skip tests that don't work on Windows
-  if (WIN32)
-    list(REMOVE_ITEM gtest_sources Converter_TEST.cc)
-    list(REMOVE_ITEM gtest_sources FrameSemantics_TEST.cc)
-    list(REMOVE_ITEM gtest_sources ParamPassing_TEST.cc)
-    list(REMOVE_ITEM gtest_sources Utils_TEST.cc)
-    list(REMOVE_ITEM gtest_sources XmlUtils_TEST.cc)
-    list(REMOVE_ITEM gtest_sources parser_urdf_TEST.cc)
+target_link_libraries(${PROJECT_LIBRARY_TARGET_NAME}
+  PUBLIC
+    ignition-math${IGN_MATH_VER}::ignition-math${IGN_MATH_VER}
+    ignition-utils${IGN_UTILS_VER}::ignition-utils${IGN_UTILS_VER}
+  PRIVATE
+    TINYXML2::TINYXML2)
+
+  if (USE_INTERNAL_URDF)
+    target_include_directories(${PROJECT_LIBRARY_TARGET_NAME} PRIVATE
+      ${CMAKE_CURRENT_SOURCE_DIR}/urdf)
+    if (WIN32)
+      target_compile_definitions(${PROJECT_LIBRARY_TARGET_NAME} PRIVATE -D_USE_MATH_DEFINES)
+    endif()
+  else()
+    target_link_libraries(${PROJECT_LIBRARY_TARGET_NAME} PRIVATE
+      IgnURDFDOM::IgnURDFDOM)
   endif()
 
-  ign_build_tests(
-    TYPE UNIT
-    SOURCES ${gtest_sources}
-    INCLUDE_DIRS
-      ${CMAKE_CURRENT_SOURCE_DIR}
-      ${PROJECT_SOURCE_DIR}/test
-  )
+if (WIN32)
```

```suggestion
if (WIN32 AND USE_INTERNAL_URDF) 
```

Unless I miss something, we do not need this if we are using the external urdfdom.

---

Yup, that makes sense. https://github.com/gazebosim/sdformat/pull/1343/commits/543730601c7bf2770948abadb108a889d303ff10